### PR TITLE
fix: preserve project folder after init failure

### PR DESCRIPTION
**Pull request by koredefashokun**
Summary:
---------

Fixes issues regarding [this comment](https://github.com/react-native-community/cli/issues/448#issuecomment-512796812) on [#448](https://github.com/react-native-community/cli).

Test Plan:
----------

Create a new project with the `init` command. It should no longer delete the project folder if the project creation process fails.

### CODE REVIEW

File: packages/cli/src/commands/init/init.js [hunk 1]
```diff
@@ -222,9 +222,6 @@ export default (async function initialize(
     printRunInstructions(projectFolder, projectName);
   } catch (e) {
     logger.error(e.message);
-    // Only remove project if it didn't exist before running `init`
-    if (!directoryExists) {
-      fs.removeSync(path.resolve(rootFolder, directoryName));
-    }
+    // Do not remove project, to allow for manual troubleshooting.
```
Comment (thymikee) [inline packages/cli/src/commands/init/init.js]: The comment is not necessary
```suggestion
```

File: packages/cli/src/commands/init/init.js [hunk 2]
```diff
@@ -222,9 +222,5 @@ export default (async function initialize(
     printRunInstructions(projectFolder, projectName);
   } catch (e) {
     logger.error(e.message);
-    // Only remove project if it didn't exist before running `init`
-    if (!directoryExists) {
```
Comment (lucasbento) [inline packages/cli/src/commands/init/init.js]: You can remove the declaration of this variable from line `216` as well.

**Comment by koredefashokun** [comment]
@lucasbento Done in the latest commit.

**Comment by thymikee** [comment]
Thank you @koredefashokun :)

**Pull request merged**